guardian/atom-workshop
# Bump Webpack to v5 and node to 20

## What does this change?

This PR bumps Webpack to v5. Quite a few related packages also needed to be bumped (e.g. Babel, various loaders). Node is also bumped to `20`.

This work started because `atom-workshop` was using node 12 - [long out of LTS](https://nodejs.dev/en/about/releases/). Dependencies that required Webpack 2 were incompatible with higher versions of node - so in order to bump Node I had to also bump Webpack (deciding to get it up-to-date rather than leaving at 3 or 4). Node 20 is in active development and will be in maintenance mode until April 2026.

The Webpack API for declaring loaders has changed from v3 to v5 so I've made some changes to the Webpack `conf` file for that reason.  I've removed `file-loader` and `url-loader` instances in favour of Webpack 5's "asset/resource" test. I've also removed `import-loader` and `export-loader`, which should sort a high priority vulnerability.

`ExtractTextPlugin` is now deprecated, so is replaced here with `MiniCssExtractPlugin`.

Jest needed to be bumped due to a package incompatibility - consequently the snapshots have needed to be updated, and a test needed to be updated as it was behaving differently after the package bump (it wasn't picking up on an error thrown asyncrously). I've also changed the arguments to our `lint` and `test` commands to apply to a more specific range of files - both were picking up files built in the `target` directory.

This PR also removes `webpack-webserver.conf.js` as it's been unused since https://github.com/guardian/atom-workshop/pull/259

I've added an `if` condition around the call to `this.props.updateFormErrors` in `ManagedEditorField` because a new React error log was happening when the function was called when it hadn't been defined - it was already not being passed as a prop in many places so I'm assuming this is to do with changes related to logging via either Babel or React version changes as the app is still functioning as before.

Along with the previous PRs, this removes all but one of the high priority client-side vulnerabilities in atom-workshop (the remaining one in is Scribe, hopefully to be replaced with `prosemirror-editor`).

![image](https://github.com/guardian/atom-workshop/assets/34686302/f18bf342-610a-462c-85d2-791b772f8877)


## How to test

Run the project locally according to the instructions in the readme. Does it build and run as expected? Do on-page assets seem to be getting bundled and loaded correctly?
